packages/lesswrong/lib/notificationTypes.tsx

```diff
@@ -164,6 +166,16 @@ export const getDocumentSummary = async (documentType: NotificationDocument | nu
         associatedUserName: userGetDisplayName(targetUser),
         displayName: null,
       }
+    case 'dialogueMatchPreference':
+      const dialogueMatchPreference = await DialogueMatchPreferences.findOne({ _id: documentId })
+      const dialogueCheckMatch = await DialogueChecks.findOne({ _id: dialogueMatchPreference?.dialogueCheckId })
+      const userMatch = await Users.findOne(dialogueCheckMatch?.userId)
```

Comment:
I would rewrite this so that we potentially passing in `undefined` for the _id selectors to `DialogueChecks.findOne` and `Users.findOne` (if either `dialogueMatchPreference` or `dialogueCheckMatch` return `null`).  That'll pull up a random document of that type.

Something like:
```
      const dialogueMatchPreference = await DialogueMatchPreferences.findOne(documentId);
      if (!dialogueMatchPreference) return null;

      const dialogueCheckMatch = await DialogueChecks.findOne(dialogueMatchPreference.dialogueCheckId);
      if (!dialogueCheckMatch) return null;

      const userMatch = await Users.findOne(dialogueCheckMatch.userId);
      if (!userMatch) return null;

      return { ... };
```

Reply:
This is more robust against any potential future changes which might allow random user's dialogue check info to slip through in notifications for this document type.